client/login.go

```diff
@@ -219,3 +220,108 @@ func RefreshToken(data RefreshTokenData) {
 	}
 	TasksMux.Unlock()
 }
+
+// OAuth2LogoutHandler handles the request signaling that the user has logged out.
+// It devalidates the
+func OAuth2LogoutHandler(w http.ResponseWriter, r *http.Request) {
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Error reading request body: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer r.Body.Close()
+
+	var data RefreshTokenData
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		http.Error(w, "Error parsing JSON: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	go OAuth2Logout(data)
+
+	w.WriteHeader(http.StatusOK)
+}
+
+// OAuth2Logout sends revocation request to the server to revoke the tokens.
+// It logs out the user from all add-ons.
+func OAuth2Logout(data RefreshTokenData) {
+	var wg sync.WaitGroup
+	var ch = make(chan error, 2)
+	wg.Add(2)
+	go RevokeOAuth2Token(data, "api_key", ch, &wg)
+	go RevokeOAuth2Token(data, "refresh_token", ch, &wg)
+	wg.Wait()
+	close(ch)
+	var errors []error
+	for err := range ch {
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	var message, status string
+	if len(errors) == 0 {
+		message = "Logout OK, tokens successfully revoked on the server"
+		status = "finished"
+	} else if len(errors) == 1 {
+		message = fmt.Sprintf("Logout partially OK, failed to revoke token: %v", errors[0])
+		status = "error"
+	} else {
+		message = fmt.Sprintf("Logout partially OK, failed to revoke tokens: %v", errors)
+		status = "error"
+	}
+
+	for appID := range Tasks {
```

Comment:
Tady jenom upozornuju, ze logout odhlasi vsechny pripojene instance. Jinak by nam totiz zustaly Blendery s nevalidnim tokenem. Plus se to chova podobne jako login, ktery take prihlasi vsude. Jen prosim o check @vilemduha, ze to takto chceme.